packages/analytics/amplify_analytics_pinpoint_dart/lib/src/analytics_plugin_impl.dart

```diff
@@ -80,9 +82,13 @@ class AmplifyAnalyticsPinpointDart extends AnalyticsPluginInterface {
   /// Storage key for the static Pinpoint endpoint id
   @visibleForTesting
   static const String endpointIdStorageKey = 'UniqueId';
+  static const String _endpointInformationVersionKey =
+      'EndpointInformationVersionKey';
   static const String _endpointGlobalAttrsKey = 'EndpointGlobalAttributesKey';
   static const String _endpointGlobalMetricsKey = 'EndpointGlobalMetricsKey';
 
+  static const String _endpointInformationCurrentVersionValue = '0';
```

Comment:
What is the purpose of this? Can you add a comment? Would an enum make more sense here?

Reply:
```
  /// Version of metadata stored into the `endpointInfoStore`
  /// If stored data changes, increment this value
  static const String _endpointInformationCurrentVersionValue = '0';

```

Added the above comments.  The initial idea is to track if we have read legacy data from native.  

If so, we store this value into secure storage to avoid further reads from legacy data.  

In case of future changes, if we need to change legacy read behavior to read other values we would increment this value and store that instead.  Having this value stored before allows us to quickly know what the storage state is (either no value, "0", "1", etc)

Reply:
Okay. I would suggest using an enum for this as we did in Auth. It provides a nice way to document this.

https://github.com/aws-amplify/amplify-flutter/blob/834a81d0192c0e2a6700dd9452e750308da60e65/packages/auth/amplify_auth_cognito_dart/lib/src/credentials/credential_store_keys.dart#L10-L20